packages/databricks-vscode/src/configuration/auth/MetadataService.ts

```diff
@@ -19,6 +19,8 @@ export class MetadataService implements Disposable {
 
     private magic!: string;
     private _apiClient: ApiClient | undefined;
+    private onDidChangeMagicEvent = new EventEmitter<void>();
+    public onDidChangeMagic = this.onDidChangeMagicEvent.event;
```

Comment:
What does `magic` refer to and how is it related to `databricks.yml`?

Reply:
The metdataservice has a nonce (`http:localhost:port/nonce`). We need to rewrite env vars whenever this nonce is regenerated. The env vars allow bundle to pickup auth from metadata-service.